src/main/java/me/lokka30/levelledmobs/listeners/paper/PlayerDeathListener.java

```diff
@@ -100,13 +100,11 @@ public boolean onPlayerDeathEvent(final @NotNull PlayerDeathEvent event) {
 
     private void updateDeathMessage(final @NotNull PlayerDeathEvent event, final @NotNull NametagResult nametagResult) {
         if (!(event.deathMessage() instanceof final TranslatableComponent tc)) {
+            // This can happen if another plugin destructively changes the death message.
             return;
         }
 
-        final String playerKilled = extractPlayerName(tc);
-        if (playerKilled == null) {
-            return;
-        }
```

Comment:
@stumper66, how come do we need to extract the player via components?

Reply:
We are replacing death message with the translated name of the mob that killed the player. To do this we have to reconstruct everything from scratch. The player name is passed as an argument into the translatable component so it will translate on the client side the proper message for how the player was killed.